scripts/ci/junit2slack/main.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/GoogleCloudPlatform/testgrid/metadata/junit"
+	"github.com/slack-go/slack"
+	"log"
+	"os"
+)
+
+func main() {
+	// We're only logging errors currently. If we log more, we should probably switch logging libraries
+	log.SetOutput(os.Stderr)
+
+	var slackAttachments []slack.Attachment
+	var junitFiles []*junit.Suites
+
+	// We should accept all file names at once since we're using `go run` to run this program. No need to recompile
+	// for each file we want to parse
+	for _, fileName := range os.Args[1:] {
+		if _, err := os.Stat(fileName); err == nil {
+			data, err := os.ReadFile(fileName)
+			if err != nil {
+				log.Printf("error while reading %s: %s", fileName, err)
+				continue
+			}
+
+			junitSuites, err := junit.Parse(data)
+			if err != nil {
+				log.Printf("error while parsing junit suites in %s: %s", fileName, err)
+				continue
+			}
+			junitFiles = append(junitFiles, junitSuites)
+
+		} else if errors.Is(err, os.ErrNotExist) {
+			log.Printf("%s doesn't exist: %s", fileName, err)
+		} else {
+			log.Printf("error while trying to find %s: %s", fileName, err)
+		}
+	}
+
+	slackMsg := convertJunitToSlack(junitFiles)
+	if slackMsg == nil {
+		log.Printf("warning: no slack message set")
+		return
+	} else {
+		slackAttachments = append(slackAttachments)
+	}
+
+	b, err := json.Marshal(slackMsg)
+	if err != nil {
+		log.Printf("error while marshaling Slack message to json: %s", err)
+	}
+	fmt.Println(string(b))
+}
+
+func convertJunitToSlack(junitFiles []*junit.Suites) []slack.Attachment {
```

Comment:
you can also do `junitFiles ...*junit.Suites`. This way when you test with just one file, you can just do `convertJunitToSlack(suites)` where `suites` is of type `*junit.Suites`